libcontainer/init_linux.go

```diff
@@ -13,6 +13,7 @@ import (
 	"strings"
 
 	"github.com/containerd/console"
+	"github.com/moby/sys/user"
```

Comment:
Just a question:
How about not depend on "github.com/moby/sys/user"?
Maybe we don't need to deprecate the entire `github.com/opencontainers/runc/libcontainer/user`, but some functions in it? I think we can keep our own `user` struct in runc?
Or else, when we are doing the next step in https://github.com/opencontainers/runc/issues/3998#issuecomment-1702005233 , if we found we need some functions not implemented in "github.com/moby/sys/user", we need to open a PR in "github.com/moby/sys/user". For example, we may need some partial codes in https://github.com/moby/sys/blob/main/user/user.go#L295-L428

Reply:
I'm open to suggestions!

My initial take on this PR was to;

- try to keep the diff small (but very much open to follow-ups, or to include additional changes)
- my intent would be to (if we agree on that) include this in a release, to get the message out to consumers "stop using this, use `github.com/moby/sys/user` instead"
- but to provide the aliases, so that things don't _break_ on "day 1" (i.e., make sure that updating the `runc` dependency "works" without changes)
- once it's in a release, we can consider removing the aliases / stubs

As to our own use; yes, that's a good question:

- what are the parts we want `runc` to be using from `moby/sys/user` (if any?) 
- _some_ of that may depend on the decision on the original proposal (i.e., should `runc` stop with resolving usernames? can we ? (without it being a huge breaking change?) can we do so in "stages" (deprecation warning -> opt-out -> opt-in -> remove feature?))
- ☝️ as long as we still _do_ the parsing, I'd prefer to use the `moby/sys/user` implementation (i.e., make sure that's the "canonical" implementation, and that we don't accidentally diverge)
- ☝️ but of course, once we _stop_ doing the parsing, we can remove that dependency (for the utilities)
- w.r.t. the _types_ used; I'd be 100% in favor of defining a local, possibly un-exported type (which may be the same, or similar to the existing type in `sys/user`); that is, assuming that no external consumer should ever have to use the type.

Reply:
> * what are the parts we want `runc` to be using from `moby/sys/user` (if any?)

Maybe about supplementary group ids?

Reply:
> * _some_ of that may depend on the decision on the original proposal (i.e., should `runc` stop with resolving usernames? can we ? (without it being a huge breaking change?) can we do so in "stages" (deprecation warning -> opt-out -> opt-in -> remove feature?))

I think `runc` has stopped resolving usernames and groupnames now?

Reply:
> * I'd be 100% in favor of defining a local, possibly un-exported type (which may be the same, or similar to the existing type in `sys/user`);

Yes, maybe this is another solution.

Thanks your explaination. I think everything could be considered in the next step.